auditbeat/module/file_integrity/eventreader_fsnotify.go

```diff
@@ -56,7 +55,15 @@ func (r *reader) Start(done <-chan struct{}) (<-chan Event, error) {
 	if err := r.watcher.Start(); err != nil {
 		return nil, errors.Wrap(err, "unable to start watcher")
 	}
-	go r.consumeEvents(done)
+
+	queueDone := make(chan struct{})
+	queueC := make(chan []*Event)
+
+	// Launch a separate goroutine to fetch all events that happen while
+	// watches are being installed.
+	go func() {
+		queueC <- r.enqueueEvents(queueDone)
```

Comment:
When the writer is done with `queueC` it should be closed.

Reply:
done